File: scenarios/fsharp-monokai/input.fs
module Demo

open System

/// Recursive factorial
let rec factorial n =
    if n <= 1 then 1
    else n * factorial (n - 1)

type Shape =
    | Circle of float
    | Rectangle of width: float * height: float

let area shape =
    match shape with
    | Circle r -> Math.PI * r * r
    | Rectangle (w, h) -> w * h

[<EntryPoint>]
let main argv =
    let nums = [1..5]
    let total = nums |> List.map (fun x -> x * x) |> List.sum
    printfn "Total: %d, fact: %d" total (factorial 5)
    0

File: scenarios/scala-catppuccin-mocha/input.scala
object Demo {
  // Recursive factorial
  def factorial(n: Int): Int =
    if (n <= 1) 1 else n * factorial(n - 1)

  sealed trait Shape
  case class Circle(r: Double) extends Shape
  case class Rect(w: Double, h: Double) extends Shape

  def area(s: Shape): Double = s match {
    case Circle(r)  => math.Pi * r * r
    case Rect(w, h) => w * h
  }

  def main(args: Array[String]): Unit = {
    val nums = List(1, 2, 3, 4, 5)
    val squares = nums.map(x => x * x)
    val name = "Scala"
    println(s"Hello, $name! sum=${squares.sum}")
  }
}
